src/cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js

```diff
@@ -27,14 +27,25 @@ server.get(
     const PaRequest = req.querystring.PaRequest;
     const MD = req.querystring.MD;
     const TermURL = URLUtils.https('Adyen-AuthorizeWithForm');
+    const signature = req.querystring.signature;
 
-    res.render('adyenform', {
-      issuerUrl: IssuerURL,
-      paRequest: PaRequest,
-      md: MD,
-      ContinueURL: TermURL,
-    });
-    next();
+    const currentSignature = AdyenHelper.getAdyenHash(
+        IssuerURL.substr(IssuerURL.length - 25),
+        MD.substr(1, 25),
+    );
+
+    if(signature === currentSignature) {
+      res.render('adyenform', {
+        issuerUrl: IssuerURL,
+        paRequest: PaRequest,
+        md: MD,
+        ContinueURL: TermURL,
+      });
```

Comment:
An easier and better solution would be to retrieve the issuerUrl, paRequest, md, not from the querystring but from the session.

Reply:
We thought about using the session but we feel it's not the most reliable. Some merchants are reporting issues that their shoppers are losing their session and their baskets are emptied, especially due to browser issues. So eventually we decided to keep the querystrings and with the signature they should be secure

Reply:
Can we investigate where else we use querystring because everywhere this is a vulnerability if we do not wrap it with a signature we have an issue.